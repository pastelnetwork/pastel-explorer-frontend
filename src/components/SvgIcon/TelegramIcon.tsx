const TelegramIcon = () => {
  return (
    <svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path d="M22.9977 3.1448C22.6238 2.83952 22.0366 2.79584 21.4295 3.03056H21.4285C20.79 3.27728 3.35378 10.4792 2.64398 10.7734C2.51487 10.8166 1.38735 11.2218 1.5035 12.1242C1.60718 12.9378 2.51338 13.2747 2.62404 13.3136L7.05686 14.7752C7.35095 15.7179 8.4351 19.196 8.67486 19.939C8.8244 20.4022 9.06815 21.0109 9.49533 21.1362C9.87017 21.2754 10.243 21.1482 10.4843 20.9658L13.1944 18.5451L17.5694 21.8307L17.6736 21.8907C17.9707 22.0174 18.2553 22.0808 18.527 22.0808C18.7368 22.0808 18.9382 22.0429 19.1306 21.967C19.7861 21.7078 20.0483 21.1064 20.0757 21.0382L23.3436 4.68128C23.543 3.80768 23.2658 3.3632 22.9977 3.1448ZM10.9663 15.3598L9.4709 19.1998L7.97552 14.3998L19.4401 6.23984L10.9663 15.3598Z" />
    </svg>
  );
};

export default TelegramIcon;
