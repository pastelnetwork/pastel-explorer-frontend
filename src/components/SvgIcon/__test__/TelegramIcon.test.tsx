import { shallow } from 'enzyme';

import TelegramIcon from '../TelegramIcon';

describe('components/SvgIcon/TelegramIcon', () => {
  const wrapper = shallow(<TelegramIcon />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <svg>', () => {
    expect(wrapper.find('svg').length).toBe(1);
  });

  test('should render <path>', () => {
    expect(wrapper.find('path').length).toBe(1);
  });
});
