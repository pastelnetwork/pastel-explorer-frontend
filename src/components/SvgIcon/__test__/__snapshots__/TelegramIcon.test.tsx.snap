// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SvgIcon/TelegramIcon renders correctly 1`] = `ShallowWrapper {}`;
